### PR TITLE
SONAR-8660 add organization to response of api/components/suggestions

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java [hunk 1]
```diff
@@ -112,33 +114,41 @@ SuggestionsWsResponse doHandle(String query) {
...
-      dbClient.closeSession(dbSession);
-    }
+  private List<String> searchInIndex(ComponentIndexQuery componentIndexQuery) {
+    return index.search(componentIndexQuery);
   }
 
-  private static Component dtoToComponent(ComponentDto result) {
+  private static Component dtoToComponent(ComponentDto result, Map<String, String> organizationKeysByUuid) {
     return Component.newBuilder()
+      .setOrganization(organizationKeysByUuid.get(result.getOrganizationUuid()))
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java]: If the organization does not exist, then request will fail with a NullPointerException. Not a pb so far as we don't expect orphans, but still a clear message would be great.

File: sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java [hunk 2]
```diff
@@ -102,9 +105,33 @@ public boolean hasRootComponentPermissions(DbSession dbSession, long rootCompone
   }
 
   public void insert(DbSession dbSession, GroupPermissionDto dto) {
+    ensureComponentPermissionConsistency(dbSession, dto);
+    ensureGroupPermissionConsistency(dbSession, dto);
     mapper(dbSession).insert(dto);
   }
 
+  private void ensureComponentPermissionConsistency(DbSession dbSession, GroupPermissionDto dto) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java]: MAJOR Make "ensureComponentPermissionConsistency" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java [hunk 3]
```diff
@@ -102,9 +105,33 @@ public boolean hasRootComponentPermissions(DbSession dbSession, long rootCompone
...
+      return;
+    }
+    ComponentMapper componentMapper = dbSession.getMapper(ComponentMapper.class);
+    checkArgument(
+      componentMapper.countComponents(dto.getOrganizationUuid(), dto.getResourceId()) == 1,
+      "Can't insert permission '%s' for component with id '%s' in organization with uuid '%s' because this component does not belong to organization with uuid '%s'",
+      dto.getRole(), dto.getResourceId(), dto.getOrganizationUuid(), dto.getOrganizationUuid());
+  }
+
+  private void ensureGroupPermissionConsistency(DbSession dbSession, GroupPermissionDto dto) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/GroupPermissionDao.java]: MAJOR Make "ensureGroupPermissionConsistency" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: sonar-db/src/main/java/org/sonar/db/permission/UserPermissionDao.java [hunk 4]
```diff
@@ -107,9 +108,21 @@ public boolean hasRootComponentPermissions(DbSession dbSession, long rootCompone
   }
 
   public void insert(DbSession dbSession, UserPermissionDto dto) {
+    ensureComponentPermissionConsistency(dbSession, dto);
     mapper(dbSession).insert(dto);
   }
 
+  private void ensureComponentPermissionConsistency(DbSession dbSession, UserPermissionDto dto) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/permission/UserPermissionDao.java]: MAJOR Make "ensureComponentPermissionConsistency" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request closed without merging**